nifi-bootstrap/src/main/java/org/apache/nifi/bootstrap/RunNiFi.java

```diff
@@ -324,27 +324,46 @@ private void registerNotificationServices(final NotificationServiceManager manag
         defaultLogger.info("Registered {} Notification Services for Notification Type {}", registered, type);
     }
 
-    File getStatusFile() {
+    File getStatusFile() throws IOException{
```

Comment:
Default method is used by the ShutdownHook class which is in the same package and allows it to not have to provide a logger. Not sure why the overloaded method is public (was introduced  early on in NIFI-145, default method came later) since it appears to only be used by RunNifi. I will research to see if the public method can really because private/protected.